tests/test_middleware.py

```diff
@@ -124,7 +125,39 @@ def test_deprecation_setting_SCRAPY_POET_OVERRIDES(settings) -> None:
 
 
 @attr.s(auto_attribs=True)
-class OptionalAndUnionPage(WebPage):
+class OptionalAndUnionPageNew(WebPage):
+    breadcrumbs: BreadcrumbsExtraction
+    opt_check_1: Optional[BreadcrumbsExtraction]
```

Comment:
Just in case, could you please add a test which checks that in case of `Optional[something]` things don't break if `something` is not available? There are similar tests in test_optional_and_unions_old, but not exactly that.

Reply:
Great catch. If it’s not *injectable*, that indeed causes an issue, I need to figure out a solution for that. It might have to do with attrs’ attribute default values not being recognized by andi, because I don’t think it happens for default values in regular callables (e.g. Scrapy callbacks).

> not available

Did you mean not injectable, or should we also support scenarios where a bad type hint is provided? (e.g. `var1: "UnexistingType` or `var2: "Optional[UnexistingType]`)

Reply:
Addressed by https://github.com/scrapinghub/andi/pull/33.